generators/app/index.js

```diff
@@ -149,6 +157,21 @@ module.exports = JhipsterGenerator.extend({
             }.bind(this));
         },
 
+        checkYarn: function () {
+            if (!this.checkInstall || this.skipClient || !this.yarnInstall) return;
+            var done = this.async();
+            exec('yarn --version', function (err) {
+                if (err) {
+                    this.warning('yarn is not found on your computer.\n',
+                        ' Using npm instead');
+                    this.yarnInstall = false;
+                } else {
+                    this.yarnInstall = true;
```

Comment:
@deepu105 : ure right, I didnt test these options: --skipServer --yarn with no yarn installed, so I think it will throw exception